src/SFA.DAS.EmployerFinance/Queries/GetAccountFinanceOverview/GetAccountFinanceOverviewQueryHandler.cs

```diff
@@ -0,0 +1,81 @@
+﻿using System;
+using System.Linq;
+using System.Threading.Tasks;
+using MediatR;
+using SFA.DAS.EmployerFinance.Models.ExpiringFunds;
+using SFA.DAS.EmployerFinance.Services;
+using SFA.DAS.NLog.Logger;
+
+namespace SFA.DAS.EmployerFinance.Queries.GetAccountFinanceOverview
+{
+    public class GetAccountFinanceOverviewQueryHandler : IAsyncRequestHandler<GetAccountFinanceOverviewQuery, GetAccountFinanceOverviewResponse>
+    {
+        private readonly IDasForecastingService _dasForecastingService;
+        private readonly IDasLevyService _levyService;
+        private readonly ILog _logger;
+
+        public GetAccountFinanceOverviewQueryHandler(
+            IDasForecastingService dasForecastingService,
+            IDasLevyService levyService, 
+            ILog logger)
+        {
+            _dasForecastingService = dasForecastingService;
+            _levyService = levyService;
+            _logger = logger;
+        }
+
+        public async Task<GetAccountFinanceOverviewResponse> Handle(GetAccountFinanceOverviewQuery query)
+        {
+            if (!query.AccountId.HasValue)
```

Comment:
Should this not be done via a validator?

Reply:
good point, no idea why i didn't use one. I'll change this now.

Reply:
Actually I do after looking at it. The validation is handled in the MVC mapping via a filter. However I'm not sure if this could be called via an internal process so need to get some feedback around that first before I change anything.

Reply:
I think that's why I didn't do fully validation on it